File: Java/AndresP/algoritmos de ordenamiento/shellsort.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

import java.util.ArrayList;

public class shellsort {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        String entrada[] = br.readLine().split(" ");
        ArrayList<Integer> entrada2=new ArrayList<Integer>();
        ArrayList<Integer> arr2=new ArrayList<Integer>();
        int arr[] = new int[entrada.length];
        for (int i = 0; i < entrada.length; i++) {
            arr2.add(entrada2.get(i));
            arr[i] = Integer.parseInt(entrada[i]);
        }
        ahellsort(arr);
        print(arr, arr.length);
    }

    static int ahellsort(int arr[]) {
        int n = arr.length;
        for (int gap = n / 2; gap > 0; gap /= 2) {
            for (int i = gap; i < n; i++) {
                int temp = arr[i];
                int j;
                for (j = i; j >= gap && arr[j - gap] > temp; j -= gap) {
                    arr[j] = arr[j - gap];
                    arr[j] = temp;

                }
            }

        }
        return 0;
    }

    static void print(int arr[], int n) {
        for (int i = 0; i < n; i++) {
            System.out.print(arr + " ");
        }
        System.out.println("");
    }
}

File: Codigos estudiantes por lenguaje/PY/Julian Silva/algoritmos de ordenamiento/shellsort.py
from sys import stdin, stdout
rl = stdin.readline
wr = stdout.write


def shellSort(arr=list, n=int):

    salto = n // 2
    while salto > 0:
        for i in range(salto, n):
            temp = arr[i]
            j = i
            while j >= salto and arr[j-salto] > temp:
                arr[j] = arr[j-salto]
                j -= salto
            arr[j] = temp
        salto //= 2


def print(arr, n):

    for i in arr:
        wr(f'{i} ')
    wr('\n')


arr = list(map(int, rl().split()))
print(arr, len(arr))
shellSort(arr, len(arr))
print(arr, len(arr))
